packages/react/src/components/Authenticator/Authenticator.tsx

```diff
@@ -66,7 +66,7 @@ export function AuthenticatorInternal({
   });
 
   const value = React.useMemo(
-    () => ({ components: { ...defaultComponents, ...customComponents } }),
```

Comment:
This `components` object felt unnecessary, we just unwrap it anyways, so I got rid of it. Also I would have liked this to do a deep merging of objects because they are nested with every part being optional, meaning although we are merging onto `defaultComponents` if `customComponents` has `SignIn: {}` it will wipe out the default `SignIn.Header` and `SignIn.Footer`. Not going to spend too much time worrying if we are going to have better component slots.